framework/include/utils/MooseUtils.h

```diff
@@ -383,6 +385,23 @@ tokenizeAndConvert(const std::string & str,
   }
   return true;
 }
+
+/**
+ * Convert supplied string to upper case.
+ */
+std::string toUpper(std::string name);
```

Comment:
We need to perform a copy because std::transform modifies the string, this just has the function do the copy rather than copy it internal. Not sure what is better, but this way is one less line.

Also, this is not a one-liner in general, because of this. Most of our methods take a std::string & so we do the following:

```
void foo(const std::string & name)
{
  std::string up(toUpper(name));
}
```

Without this utility:
```
void foo(const std::string & name)
{
  std::string up(name);
  std::transform(up.begin(), up.end(), ::touppper);
}
```

This class (currently) and with these changes does this alot, so I figured wrapping it up was worth it.